tests/Remove-DbaAgentJob.Tests.ps1

```diff
@@ -0,0 +1,77 @@
+        $null = Remove-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob
+        It "Should have deleted job: dbatoolsci_testjob" {
+            (Get-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob) | Should BeNullOrEmpty
+        }
+        It "Should have deleted schedule: dbatoolsci_daily" {
+            (Get-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_daily) | Should BeNullOrEmpty
+        }
+        It "Should have deleted history: dbatoolsci_daily" {
+            (Get-DbaAgentJobHistory -SqlInstance $script:instance2 -Job dbatoolsci_testjob) | Should BeNullOrEmpty
+        }
+    }
+    Context "Command removes job but not schedule" {
+        BeforeAll {
+            $null = New-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_weekly -FrequencyType Weekly -FrequencyInterval Everyday -Force
+            $null = New-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_schedule -Schedule dbatoolsci_weekly
+            $null = New-DbaAgentJobStep -SqlInstance $script:instance2 -Job dbatoolsci_testjob_schedule -StepId 1 -StepName dbatoolsci_step1 -Subsystem TransactSql -Command 'select 1'
+        }
+        AfterAll {
+            if (Get-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_weekly) { Remove-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_weekly }
+        }
+        $null = Remove-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_schedule -KeepUnusedSchedule
+        It "Should have deleted job: dbatoolsci_testjob_schedule" {
+            (Get-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_schedule) | Should BeNullOrEmpty
+        }
+        It "Should not have deleted schedule: dbatoolsci_weekly" {
+            (Get-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_weekly) | Should Not BeNullOrEmpty
+        }
+    }
+    Context "Command removes job but not history" {
+        BeforeAll {
+            $job = New-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_history
+            $null = New-DbaAgentJobStep -SqlInstance $script:instance2 -Job dbatoolsci_testjob_history -StepId 1 -StepName dbatoolsci_step1 -Subsystem TransactSql -Command 'select 1'
+            $null = Start-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_history
+        }
+    
+        $null = Remove-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_history -KeepHistory
+        It "Should have deleted job: dbatoolsci_testjob_history" {
+            (Get-DbaAgentJob -SqlInstance $script:instance2 -Job dbatoolsci_testjob_history) | Should BeNullOrEmpty
+        }
+        It "Should not have deleted history: dbatoolsci_testjob_history" {
+            (Invoke-SqlCmd2 -ServerInstance $script:instance2 -Database msdb -Query "select 1 from sysjobhistory where job_id = '$($job.jobid)'") | Should Not BeNullOrEmpty
```

Comment:
I don't know other folk's thinking on this but I try to stay clear of using this command in test, because it is one that can change and end up causing the test to break.

I would recommend just using the `$server.Databases['msdb'].Query()` method or just `$server.Query($sql,"msdb")`.